compatibility/common/src/test/java/org/projectnessie/tools/compatibility/internal/TestNessieCompatibilityExtensions.java

```diff
@@ -258,6 +298,32 @@ void never() {
     }
   }
 
+  @ExtendWith({OlderNessieServersExtension.class, SoftAssertionsExtension.class})
+  static class ApiEndpointServerSample {
+    @InjectSoftAssertions protected SoftAssertions soft;
+
+    @NessieAPI NessieApiV1 api;
+    @NessieAPI static NessieApiV1 apiStatic;
+    @NessieBaseUri URI uri;
+    @NessieBaseUri static URI uriStatic;
+    @NessieVersion Version version;
+    @NessieVersion static Version versionStatic;
+
+    static final List<Version> allVersions = new ArrayList<>();
+
+    @Test
+    void testSome() throws Exception {
+      soft.assertThat(api).isNotNull().isSameAs(apiStatic);
+      soft.assertThat(uri).isNotNull().isEqualTo(uriStatic);
+      soft.assertThat(version).isNotNull().isEqualTo(versionStatic);
+      allVersions.add(version);
+
+      // use api
+      List<Reference> references = api.getAllReferences().get().getReferences();
+      soft.assertThat(references).isNotEmpty();
```

Comment:
note that without `assertNoFailedTestEvents` the outer assertion for `allVersions` would pass just fine, even though the API usage causes an error